security/fma-generic-hardfork.md

```diff
@@ -15,49 +15,52 @@
+- **Risk Assessment:** Medium severity as no funds are at risk, though it is a full liveness failure that would be messy to recover from and require releasing and distributing an updated configuration and/or release. Likelihood is low, especially if we use the mechanics for dynamically upgrading L2 contracts established in the Ecotone hardfork: namely, inserting appropriate deposit transactions in the activation block.
 
-- **Mitigations:** We have implemented extensive unit and end-to-end testing of the activation flow.  We will be testing the activation as well on our devnets and testnets.
+- **Mitigations:** 
+
+- [ ] ACTION ITEM (BLOCKING): We have implemented extensive unit and end-to-end testing of the activation flow.
+- [ ] ACTION ITEM (BLOCKING): We will be testing the activation on our devnets and testnets.
 
-- **Detection:** Detection is straightforward as the chain will stop producing blocks.
+- **Detection:** Detection is straightforward as the chain will stop producing blocks. On OP Mainnet, P1 alarms are triggered and on-call engineers are paged if the unsafe head does not increase for 1 minute or if the safe head does not increase for 15 minutes.
 
-- **Recovery Path(s)**: Would not require a vote or hardfork, but we’d likely have to coordinate a chain config update that pushed back the date of the upgrade, and allowed node operators to rollback any bad blocks.
+- **Recovery Path(s)**: Would not require a vote or hardfork, but we’d likely have to coordinate a chain config update that pushed back the date of the upgrade, and allowed node operators to rollback any bad blocks. Estimated sequencer downtime is 30 min in a worst-case scenario where we have to reset the chain back to a block before the activation and disable the hardfork activation. Additional steps would be required from infra providers to get back to the healthy chain. They would either need to restart their op-node and op-geth with activation override command line flags/env var, or their images to an emergency release with activation disabled.
 
-    - We should also prepare datadir backups close before the upgrade, so we can use these in an emergency to rollback.
+    - [ ] ACTION ITEM (BLOCKING): We have prepared datadir backups close before the upgrade, so we can use these in an emergency to rollback.
 
-    - Estimated sequencer downtime is 30 min in a worst-case scenario where we have to reset the chain back to a block before the activation and disable the hardfork activation. Additional steps would be required from infra providers to get back to the healthy chain. They would either need to restart their op-node and op-geth with activation override command line flags/env var, or their images to an emergency release with activation disabled.
+    - [ ] ACTION ITEM (non-BLOCKING): We have prepared an emergency release with activation disabled.
```

Comment:
```suggestion
    - [ ] ACTION ITEM (non-BLOCKING): We have prepared an emergency release with activation disabled.
```
I'm not sure about this one, I feel like the overrides are a much quicker/easier recovery path, and we would probably just disable activation in the release that fixes any node software bugs?

Happy for more input on this though.

Reply:
I agree. The override is easier and makes more sense as a recovery path.